demos/common/samples/args_helper.hpp

```diff
@@ -85,3 +85,54 @@ void parseInputFilesArguments(std::vector<std::string> &files) {
         readInputFilesArguments(files, args.at(i));
     }
 }
+
+std::vector<std::string> split(const std::string &s, char delim) {
+    std::vector<std::string> result;
+    std::stringstream ss(s);
+    std::string item;
+
+    while (getline(ss, item, delim)) {
+        result.push_back(item);
+    }
+    return result;
+}
+
+std::vector<std::string> parseDevices(const std::string& device_string) {
+    std::string comma_separated_devices = device_string;
+    if (comma_separated_devices.find(":") != std::string::npos) {
+        comma_separated_devices = comma_separated_devices.substr(comma_separated_devices.find(":") + 1);
```

Comment:
The colon is searched for twice. You ought to save its index to a variable.

Reply:
updated